lib/Common/CommonDefines.h

```diff
@@ -110,6 +110,17 @@
 #define ENABLE_UNICODE_API 1                        // Enable use of Unicode-related APIs
 #endif
 
+// ICU Configuration
+#if defined(INTL_ICU) || defined(HAS_REAL_ICU)
+#define U_SHOW_CPLUSPLUS_API 0
+#define U_DISABLE_RENAMING 1
+#define U_DEFAULT_SHOW_DRAFT 0
+#define U_HIDE_DRAFT_API 1
+#define U_HIDE_DEPRECATED_API 1
+#define U_HIDE_OBSOLETE_API 1
+#define U_HIDE_INTERNAL_API 1
```

Comment:
LGTM

Reply:
@jefgen the Linux build is failing here because uloc_toUnicodeLocale* is marked as "stable" in ICU 54 but is actually U_DRAFT. Is that valid? If so, I may only define these U_HIDEs on Windows to allow us to keep using system ICU in Ubuntu 16.04 (ICU 55)

Reply:
It looks like the API was promoted to stable in ICU version 56.
Looking at the source for ICU version 54, I don't see it marked as "stable" in the official ICU sources though. It is marked as "@draft" and also "U_DRAFT"
http://source.icu-project.org/repos/icu/icu/tags/release-54-1/source/common/unicode/uloc.h

Perhaps the copy of ICU 54 on your Linux build machines is slightly modified...? What does the header file `common\unicode\uloc.h` look like?